### PR TITLE
TCO calculator

### PR DESCRIPTION
**Pull request by sowasred2012**
## Done

Added /tco-calculator, based on [copy doc](https://docs.google.com/document/d/1n3ExZwFVlw5LyAp7O3CgPthVgx8yRX5-Wg_gBQLf73c/edit#), the [design](https://github.com/canonical-web-and-design/web-squad/issues/2442), and formulas from [the spreadsheet](https://docs.google.com/spreadsheets/d/1qOFmJ2zOy0MNqANk1IQnrBgOPhLxwoj7g8Nvyl4y_Ic/edit#gid=0).

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8001/tco-calculator
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- See that the page matches the copy doc, the formulas work, and the design looks good.

## Issue / Card

Fixes https://github.com/canonical-web-and-design/web-squad/issues/2442

### CODE REVIEW

**Comment by SirSamTumless** [comment]
@sowasred2012 Is there a demo link for this?

**Comment by SirSamTumless** [comment]
@sowasred2012 a few quick things:
- The managed "Yearly operational costs" should not change when a selecting Advanced, Standard and Essential.
- There should be a clear indication that "Data volume" is measured in TB
- Can we check the "OpenStack" by default 
- Can we check the "Reference architecture" by default
- Can we check the "Advanced" support by default?
- Can we make the URL just  https://ubuntu.com/tco-calculator
- Can we move Initial rollout and yearly operational cost right below the manged and self-managed headers?

**Comment by SirSamTumless** [comment]
The "Managed" yearly operational costs seem to be missing an additional $900 for three managed MAAS nodes. Can we add $900 to the managed section and add the text *** Includes Managed MAAS.

File: static/js/src/tco-calculator.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      updateTotals();
+    });
+  });
+}
+
+function calculateStorageCost(serviceLevel, dataVolume) {
+  let tier = "tier_";
+  let adjustment = 0;
+
+  if (dataVolume <= 150) {
```
Comment (deadlight) [inline static/js/src/tco-calculator.js]: Might this be neater as a switch/case?

File: templates/tco-calculator/index.html [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+      <h2 class="p-heading--three">
+        Cloud type
+        <span class="p-tooltip--top-center" aria-describedby="cloud-type-tooltip">
+          <i class="p-icon--question"></i>
+          <span class="p-tooltip__message" role="tooltip" id="cloud-type-tooltip">A desired cloud type: OpenStack, Kubernetes or both.<br />In case only one option is selected, it is assumed that the cloud will run directly on bare-metal machines.<br />If both are selected, it is assumed that Kubernetes will run on top of OpenStack.</span>
+        </span>
+      </h2>
+
+      <div>
+        <input type="checkbox" name="cloud-type" id="ct-openstack" class="js-tco-calulator__checkbox" value="openstack" checked="">
```
Comment (deadlight) [inline templates/tco-calculator/index.html]: "calculator"

File: templates/tco-calculator/index.html [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+      <h2 class="p-heading--three">
+        Deployment type
+        <span class="p-tooltip--top-center" aria-describedby="dplmt-type-tooltip">
+          <i class="p-icon--question"></i>
+          <span class="p-tooltip__message" role="tooltip" id="dplmt-type-tooltip">Private cloud deployment can be either based on the reference architecture<br />or a custom one. Use the existing deployment option to just check<br />the pricing of support or fully managed services.</span>
+        </span>
+      </h2>
+
+      <div>
+        <input type="radio" name="deployment-type" id="dplmt-reference" class="js-tco-calulator__radio" value="reference" checked="checked">
```
Comment (deadlight) [inline templates/tco-calculator/index.html]: "calculator"

File: templates/tco-calculator/index.html [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+        <label for="dplmt-reference" class="is-h4">
+          <span class="p-heading--four">Reference architecture</span>
+          <span class="p-calculator__checkbox-label-copy">Canonical deploys the cloud base on the reference architecture.</span>
+        </label>
+      </div>
+
+      <div>
+        <input type="radio" name="deployment-type" id="dplmt-custom" class="js-tco-calulator__radio" value="custom">
```
Comment (deadlight) [inline templates/tco-calculator/index.html]: "calculator"

File: templates/tco-calculator/index.html [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+        <label for="dplmt-custom" class="is-h4">
+          <span class="p-heading--four">Custom architecture</span>
+          <span class="p-calculator__checkbox-label-copy">Canonical designs and deploys the cloud according to customer’s requirements.</span>
+        </label>
+      </div>
+
+      <div>
+        <input type="radio" name="deployment-type" id="dplmt-existing" class="js-tco-calulator__radio" value="existing">
```
Comment (deadlight) [inline templates/tco-calculator/index.html]: "calculator"

File: templates/tco-calculator/index.html [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+          <p class="p-heading--three" id="yearly-cost--self">$0</p>
+        </div>
+      </div>
+
+      <hr />
+
+      <p>Self-managed option may be better for organisations which already have knowledge, experience and resources required to operate the cloud. In this case support services should be considered as described below:</p>
+
+      <div>
+        <input type="radio" name="self-managed" id="sm-advanced" class="js-tco-calulator__radio" value="advanced" checked="checked">
```
Comment (deadlight) [inline templates/tco-calculator/index.html]: "calculator"

File: templates/tco-calculator/index.html [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+        <label for="sm-advanced" class="is-h4">
+          <span class="p-heading--four">Advanced</span>
+          <span class="p-calculator__checkbox-label-copy">The most comprehensive package including 24*7 support, production-grade SLAs, Kernel Livepatch service, Landscape service, security patches and more.</span>
+        </label>
+      </div>
+
+      <div>
+        <input type="radio" name="self-managed" id="sm-standard" class="js-tco-calulator__radio" value="standard">
```
Comment (deadlight) [inline templates/tco-calculator/index.html]: "calculator"

File: templates/tco-calculator/index.html [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+        <label for="sm-standard" class="is-h4">
+          <span class="p-heading--four">Standard</span>
+          <span class="p-calculator__checkbox-label-copy">Includes 24*5 support, less demanding SLAs, Kernel Livepatch service, Landscape service, security patches and more. Half of the cost of the Advanced package.</span>
+        </label>
+      </div>
+
+      <div>
+        <input type="radio" name="self-managed" id="sm-essential" class="js-tco-calulator__radio" value="essential">
```
Comment (deadlight) [inline templates/tco-calculator/index.html]: "calculator"

File: templates/tco-calculator/index.html [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+        <label for="sm-essential" class="is-h4">
+          <span class="p-heading--four">Essential</span>
+          <span class="p-calculator__checkbox-label-copy">The cheapest package for the least demanding organisations. Includes Kernel Livepatch service, Landscape service, security patches and more.</span>
+        </label>
+      </div>
+
+      <div>
+        <input type="radio" name="self-managed" id="sm-none" class="js-tco-calulator__radio" value="none">
```
Comment (deadlight) [inline templates/tco-calculator/index.html]: "calculator"

File: templates/tco-calculator/index.html [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+      <div>
+        <input type="checkbox" name="cloud-type" id="ct-openstack" class="js-tco-calulator__checkbox" value="openstack" checked="">
+        <label for="ct-openstack" class="is-h4">
+          <span class="p-heading--four">OpenStack</span>
+          <span class="p-calculator__checkbox-label-copy">OpenStack is an open source cloud platform providing virtual machines on demand.</span>
+        </label>
+      </div>
+
+      <div>
+        <input type="checkbox" name="cloud-type" id="ct-k8s" class="js-tco-calulator__checkbox" value="kubernetes">
```
Comment (deadlight) [inline templates/tco-calculator/index.html]: "calculator"

**Comment by spencerbygraves** [comment]
@sowasred2012 Can we right align the figures please.

**Pull request merged**